src/services/summaryHeatmapService.ts

```diff
@@ -129,8 +130,14 @@ export function getSecondaryHeatmapData(yAxis: HierarchicalItem[], connections:
 export function summaryFilterKnowledgeStatements(knowledgeStatements: Record<string, KnowledgeStatement>, summaryFilters: SummaryFilters): Record<string, KnowledgeStatement> {
 	const phenotypeIds = summaryFilters.Phenotype.map(option => option.id);
 	const nerveIds = summaryFilters.Nerve.map(option => option.id);
-	console.log(phenotypeIds, nerveIds);
-	return knowledgeStatements;
+	return Object.entries(knowledgeStatements).reduce((filtered, [id, ks]) => {
+		const phenotypeMatch = !phenotypeIds.length || phenotypeIds.includes(ks.phenotype);
+		const nerveMatch = !nerveIds.length || ks.via?.some(via => via.anatomical_entities.map(entity => entity.id).some(id => nerveIds.includes(id)));
```

Comment:
NOTE: even though the connections/KS may have vias it may not be filtered through them. Since filters shown in the dropdown are only MAJOR NERVES. This can be seen in the video above in PR description. 

cc @afonsobspinto @ddelpiano